src/transactions/types.ts

```diff
@@ -278,11 +278,11 @@ export type InputSimulateTransactionData = {
   /**
    * For a single signer transaction
    */
-  signerPublicKey: PublicKey;
+  signerPublicKey?: PublicKey;
   /**
    * For a fee payer or multi-agent transaction that requires additional signers in
```

Comment:
please add some more explanation on why/when a public key can be undefined

Reply:
Added some explanation.